packages/indexer-agent/src/utils/balance-monitor.ts

```diff
@@ -0,0 +1,39 @@
+import { Gauge } from 'prom-client'
+import { Wallet, utils } from 'ethers'
+
+import { Logger, Metrics, timer } from '@graphprotocol/common-ts'
+import { indexerError, IndexerErrorCode } from '@graphprotocol/indexer-common'
+
+interface EthBalanceMetrics {
+  operatorEthBalance: Gauge<string>
+}
+
+const registerMetrics = (metrics: Metrics): EthBalanceMetrics => ({
+  operatorEthBalance: new metrics.client.Gauge({
+    name: 'indexer_agent_operator_eth_balance',
+    help: 'Amount of ETH in the operatow wallet; a low amount could cause transactions to fail',
+    registers: [metrics.registry],
+  }),
+})
+
+export async function monitorEthBalance(
+  logger: Logger,
+  wallet: Wallet,
+  metrics: Metrics,
+): Promise<void> {
+  logger.info('Monitor operator ETH balance (refreshes every 120s)')
+
+  const balanceMetrics = registerMetrics(metrics)
+
+  timer(120_000).pipe(async () => {
+    try {
+      const balance = await wallet.getBalance()
+      const eth = parseFloat(utils.formatEther(balance))
+      balanceMetrics.operatorEthBalance.set(eth)
```

Comment:
could there be an informative log message about how much is the balance? maybe when it is somewhat low?

Reply:
I've added a log message that includes the ETH balance. What is low depends on a number of factors (e.g. current gas price, average transaction cost) and alerts for what is low may depend from indexer to indexer, who may have different preferences for how early to top up. So I'd prefer to leave that decision to them.